README.rst

```diff
@@ -93,6 +93,89 @@ Pika provides the following adapters
 - TornadoConnection  - adapter for use with the Tornado IO Loop http://tornadoweb.org
 - TwistedConnection  - adapter for use with the Twisted asynchronous package http://twistedmatrix.com/
 
+Connection recovery
+-------------------
+
+Some RabbitMQ clients using automated connection recovery mechanisms to
+reconnect and recover channels and consumers in case of network errors.
+
+Connection recovery is mostly valuable in case of consumers, because they are
+long-running tasks doing smaller pieces of work on each message, hence they
+can be restarted on errors.
+
+Different connection adapters alow you to set up different approaches to
+connection recovery.
+
+For BlockingConnection adapter exception handling can be used to check for
+connection errors. Simplified example:
+
+.. code :: python
+    import pika
+    while(True):
+    try:
+        connection = pika.BlockingConnection(parameters)
+        channel = connection.channel()
+        channel.basic_consume('standard', on_message_callback)
+        channel.start_consuming()
+    # Do not recover if connection was closed by broker
+    except pika.exceptions.ConnectionClosedByBroker:
+        break
+    # Do not recover on channel errors
+    except pika.exceptions.AMQPChannelError:
+        break
+    # Recover on all other connection errors
+    except pika.exceptions.AMQPConnectionError:
+        continue
+
+This example can be found in `examples/consume_recover.py`.
```

Comment:
I wonder if it's possible to instruct the documentation system to pull the source code directly out of `examples/consume_recover.py` instead of maintaining duplicates?